File: swift/입문/모음_제거.swift
import Foundation

func solution(_ my_string:String) -> String {
    guard (1...1000).contains(my_string.count),
      my_string.allSatisfy({ ($0 >= "a" && $0 <= "z") || $0 == " " }) else { return "" }
    return my_string.filter { !["a","e","i","o","u"].contains($0) }
}

// import Foundation

// func solution(_ my_string:String) -> String {
//     guard (1...1000).contains(my_string.count),
//       my_string.allSatisfy({ ($0 >= "a" && $0 <= "z") || $0 == " " }) else {
//     return ""
// }
//     return Array(my_string).filter { !["a","e","i","o","u"].contains($0) }.map { String($0) }.joined(separator: "")
// }

// filter 반환, swift는 in 없음(파이썬이랑 헷갈림)

File: python/입문/모음_제거.py
# 다시해보기
# def solution(my_string):
#     answer = "".join([s for s in my_string if not (s in "aeiou")])
#     return answer

def solution(my_string):
    answer = "".join(list(filter(lambda s: s not in ["a","e","i","o","u"],my_string)))
    return answer